src/test/moves/thunder_wave.test.ts

```diff
@@ -0,0 +1,122 @@
+    game = new GameManager(phaserGame);
+    game.override.battleType("single");
+    game.override.starterSpecies(Species.PIKACHU);
+    game.override.moveset([Moves.THUNDER_WAVE]);
+    game.override.enemyMoveset([Moves.SPLASH, Moves.NONE, Moves.NONE, Moves.NONE]);
+    vi.spyOn(allMoves[Moves.THUNDER_WAVE], "accuracy", "get").mockReturnValue(100);
+  });
+
+  // References: https://bulbapedia.bulbagarden.net/wiki/Thunder_Wave_(move)
+
+  test(
+    "Thunder Wave as-is paralyzes non-statused Pokemon that are not Ground types",
+    async () => {
+      game.override.enemySpecies(Species.MAGIKARP);
+      await game.startBattle();
+
+      const enemyPokemon: EnemyPokemon = game.scene.getEnemyPokemon()!;
+
+      game.doAttack(getMovePosition(game.scene, 0, Moves.THUNDER_WAVE));
+      await game.phaseInterceptor.to(TurnEndPhase);
+
+      expect(enemyPokemon.status?.effect).toBe(StatusEffect.PARALYSIS);
+    },
+    TIMEOUT
+  );
+
+  test(
+    "Thunder Wave as-is does not paralyze if the Pokemon is a Ground-type",
+    async () => {
+      game.override.enemySpecies(Species.DIGLETT);
+      await game.startBattle();
+
+      const enemyPokemon: EnemyPokemon = game.scene.getEnemyPokemon()!;
+
+      game.doAttack(getMovePosition(game.scene, 0, Moves.THUNDER_WAVE));
+      await game.phaseInterceptor.to(TurnEndPhase);
+
+      expect(enemyPokemon.status).toBeUndefined();
+    },
+    TIMEOUT
+  );
+
+  test(
+    "Thunder Wave does not paralyze if the Pokemon is already status-ed",
+    async () => {
+      game.override.enemySpecies(Species.MAGIKARP);
+      await game.startBattle();
+
+      const enemyPokemon: EnemyPokemon = game.scene.getEnemyPokemon()!;
+      enemyPokemon.status = new Status(StatusEffect.BURN);
+
+      game.doAttack(getMovePosition(game.scene, 0, Moves.THUNDER_WAVE));
+      await game.phaseInterceptor.to(TurnEndPhase);
+
+      expect(enemyPokemon.status?.effect).not.toBe(StatusEffect.PARALYSIS);
+    },
+    TIMEOUT
+  );
+
+  test(
+    "Thunder Wave's effectiveness is based on its type",
+    async () => {
+      game.override.ability(Abilities.NORMALIZE);
+      game.override.enemySpecies(Species.DIGLETT);
+      await game.startBattle();
+
+      const enemyPokemon: EnemyPokemon = game.scene.getEnemyPokemon()!;
+
+      game.doAttack(getMovePosition(game.scene, 0, Moves.THUNDER_WAVE));
+      await game.phaseInterceptor.to(TurnEndPhase);
+
+      expect(enemyPokemon.status?.effect).toBe(StatusEffect.PARALYSIS);
+    },
+    TIMEOUT
+  );
+
+  test(
+    "Thunder Wave respects type immunities",
+    async () => {
+      game.override.ability(Abilities.NORMALIZE);
+      game.override.enemySpecies(Species.HAUNTER);
+      await game.startBattle();
+
+      const enemyPokemon: EnemyPokemon = game.scene.getEnemyPokemon()!;
+
+      game.doAttack(getMovePosition(game.scene, 0, Moves.THUNDER_WAVE));
+      await game.phaseInterceptor.to(TurnEndPhase);
+
+      expect(enemyPokemon.status).toBeUndefined();
```

Comment:
```suggestion
      expect(enemyPokemon.status).toBeUndefined();
      game.doAttack(getMovePosition(game.scene, 0, Moves.THUNDER_WAVE));
      await game.phaseInterceptor.to(TurnEndPhase);

      expect(enemyPokemon.status).toBeUndefined();
```
Might be safe to expect status to be undefined before as well. This is applicable to all the other tests as well.

Reply:
I think that even though this is a relatively small check + only adds a few milliseconds to the test, I think that because these Pokemon do not have items (thanks to earlier work in #3358) + the opponent Pokemon do not have abilities or passives that prevent Paralysis / lead to early recovery from Paralysis, the check is a little extra. I know its only a few ms but I am trying to be more efficient in my unit tests.